src/main/java/org/jboss/modules/ResourceLoaders.java

```diff
@@ -75,10 +77,21 @@ public static IterableResourceLoader createIterableFileResourceLoader(final Stri
      * @param jarFile the backing JAR file
      * @return the resource loader
      */
+    public static ResourceLoader createJarResourceLoader(final String name, final Path jarFile) throws IOException {
+        return new JarResourceLoader(name, jarFile, AccessController.getContext());
+    }
+
+    @Deprecated
     public static ResourceLoader createJarResourceLoader(final String name, final JarFile jarFile) {
-        return new JarFileResourceLoader(name, jarFile);
```

Comment:
sure, we need to do that before moving to this. 
From my tests new approach works much better when you have lots of deployments and there is lots of concurrency going on.

I also know what is the reason for slow down you see, am looking into finding best solution for it.